fastapi_route_logger_middleware/__init__.py

```diff
@@ -14,9 +15,15 @@ def __init__(
         *,
         logger: typing.Optional[logging.Logger] = None,
         skip_routes: typing.List[str] = None,
+        skip_regexes: typing.List[str] = None,
     ):
         self._logger = logger if logger else logging.getLogger(__name__)
         self._skip_routes = skip_routes if skip_routes else []
+        self._skip_regexes = (
+            list(map(lambda regex: re.compile(regex), skip_regexes))
```

Comment:
Compile on instantiation, not per endpoint hit. Avoids any slowdown with compilation and will throw errors long before actually used in an app to handle a request.